html/browsers/windows/auxiliary-browsing-contexts/resources/close-opener.html

```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<meta charset="utf-8">
+<html>
+<body>
+<p>This window should close its opener.</p>
+<script>
+function closeOpener () {
+  window.addEventListener('storage', e => {
+    if (e.key === 'closeAll') {
+      window.close();
+    }
+  });
+  if (window.opener) {
+    window.opener.close();
+
+    // Give the browsing context a chance to dispose of itself
+    function waitForContextDiscard () {
+      if (window.opener === null) {
+        return localStorage.setItem('openerIsNull', 'true');
+      }
+      return setTimeout(waitForContextDiscard);
```

Comment:
@zcorpan Yeah, it had a gross feel to it, for sure, and I put my head together with @jugglinmike a bit over it. In my non-scientific tests I never observed any of my browsers taking more than one loop for the discard to happen but if there is a good pattern to drop in here, I'm all ears. @jugglinmike even had the cool (but alas not successful) idea of trying to use a property descriptor to watch for when `window.opener` gets `set` to `null`...There's no evident event I can listen for: `unload` happens too soon, of course, (before the window is discarded) and I can't think of another way to verify that the window is _gone_.